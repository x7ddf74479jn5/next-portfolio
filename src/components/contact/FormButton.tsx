import clsx from "clsx";
import styles from "src/styles/components/contact/FormButton.module.scss";

import type { ButtonType } from "../../types/utils/index";

type Props = ButtonType & {
  buttonType: "apply" | "cancel";
  label?: string;
};

const FormButton: React.FC<Props> = ({ children, buttonType, label, disabled = false, ...rest }) => {
  return (
    <button
      className={clsx({
        [styles.btnApply]: buttonType === "apply",
        [styles.btnCancel]: buttonType === "cancel",
      })}
      disabled={disabled}
      {...rest}
    >
      {children || label}
    </button>
  );
};

export default FormButton;
