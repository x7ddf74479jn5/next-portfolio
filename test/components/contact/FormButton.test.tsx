import type { RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import FormButton from "src/components/contact/FormButton";

describe("FormButton", () => {
  let renderResult: RenderResult;
  let buttonElement: HTMLElement;
  const mockHandleClick = jest.fn();
  describe("apply", () => {
    beforeEach(() => {
      renderResult = render(<FormButton buttonType="apply" label="test" onClick={mockHandleClick} />);
      buttonElement = renderResult.getByRole("button");
    });

    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      expect(buttonElement).toHaveTextContent("test");
      expect(buttonElement).toBeEnabled();
      expect(buttonElement).toHaveClass("btnApply");
    });

    it("user interaction", () => {
      userEvent.click(buttonElement);
      expect(mockHandleClick).toHaveBeenCalled();
    });
  });

  describe("cancel", () => {
    beforeEach(() => {
      renderResult = render(<FormButton buttonType="cancel" label="test" onClick={mockHandleClick} />);
    });

    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      const buttonElement = renderResult.getByRole("button");
      expect(buttonElement).toHaveTextContent("test");
      expect(buttonElement).toBeEnabled();
      expect(buttonElement).toHaveClass("btnCancel");
    });
  });
});
