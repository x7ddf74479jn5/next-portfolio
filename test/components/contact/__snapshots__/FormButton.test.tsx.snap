// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormButton apply matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="btnApply"
  >
    test
  </button>
</DocumentFragment>
`;

exports[`FormButton cancel matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="btnCancel"
  >
    test
  </button>
</DocumentFragment>
`;
